Objects/dictobject.c

```diff
@@ -5146,15 +5192,11 @@ dictitems_contains(_PyDictViewObject *dv, PyObject *obj)
         return 0;
     key = PyTuple_GET_ITEM(obj, 0);
     value = PyTuple_GET_ITEM(obj, 1);
-    found = PyDict_GetItemWithError((PyObject *)dv->dv_dict, key);
-    if (found == NULL) {
-        if (PyErr_Occurred())
-            return -1;
-        return 0;
+    result = PyDict_GetItemRef(dv->dv_dict, key, &found);
+    if (result == 1) {
```

Comment:
Or maybe even `value != NULL`?

Reply:
Well, I adapted your suggested ``(result > 0)`` code.

Testing ``result == 1`` and ``value != NULL`` is exactly the same: if it's not the case, it's a big bug in the function implementation. Here since we store the function result in a variable, I like to reuse it.

Micro-optimization note. Also, maybe comparing a "small" int is faster than comparing a "large" pointer :-) Especially because function result is usually stored in a common "fast" CPU register :-)